### ISSUE TITLE
Ability to give a randomized payment between two values

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#3743]
Currently, payments for missions are either a fixed value or a value based off of the cargo and passengers transported between the quickest non-wormhole hyperdrive route from source to destination. This works fine for the types of missions in game, but having the ability to give a randomized payment between two values (or perhaps even a randomized payment above a single value where further values from the base payment become more and more rare) could open the possibility to more missions. I know that @Pointedstick has had ideas for a gambling mission with pirates, and such a randomized payment method would be helpful. I've also mentioned in the past (https://github.com/endless-sky/endless-sky/issues/3139) possible missions where one finds treasure on a pirate ship; a randomized payment attribute could be used for this to randomize the amount of credits one finds on a pirate ship. One might even encounter a character with a nack for having trouble providing payment; he/she might say you'll get paid 100,000 credits, but at the destination he/she might only have somewhere between 70k and 90k credits to give (and randomizing this would mean that players could pick up on this when they do the mission on multiple pirates).

**Comment by jafdy** [comment]
Might be a good idea to add a step of values so you don't end up with "... and 2 credits". So `payment random 70000 90000 1000` would be a payment between 70k and 90k in steps of 1k (so you can only get complete thousands).

**Comment by gunqqer** [comment]
Just to clarify, `payment random <low> <high> <step>` is your suggestion,
jafdy ? If so, +1

**Comment by jafdy** [comment]
Yes, that is what I wrote. It could be done in a different way though.

**Comment by Janaszar** [comment]
In the case for missions where you get a flat fee and a variable fee, such as Bulk Transportation Jobs, would this just be a change to the flat fee?

I think it might be nicer to have the option of changing the Payment formula's coefficients, such as...
base + multiplier * (jumps + 1) * (Gamma * (Alpha * # of passengers) + (Beta * tons of cargo))
So the Payment function could be... (I think I'm doing the brackets right...)
payment [<base> [<base1>[<probability>]] [<multiplier> [<Alpha> [<Alpha1>[<probability>]] [<Beta>[<Beta1>[<probability>]][<Gamma>[<Gamma1>[<probability>]]]]]]]

This way, we can include the options of randomizing using the negative binomial function already used in cargo specification, allowing for rare high-profit jobs.

As for the stepping, I'd be fine with "continuous" output.  I don't see much of a benefit to having the stepping since payments don't have that anyway.
